### ISSUE TITLE
parameter "logprob": the token with highest score is not the one chosen to filled in answer

### ISSUE THREAD

**Issue by kakaxisisan** [vllm-project/vllm#2214]
![image](https://github.com/vllm-project/vllm/assets/34512704/88e00b63-f947-494b-85d7-6deb3dee6558)
as show in picture, 151749 with score -0.06xxx and 101158 with -3.25 are the token in answer(token_ids)，but in the position of 151749, the 151750 has smaller score and in verse for 101158.

**Comment by WoosukKwon** [comment]
Hi @kakaxisisan, what temperature did you use for generation? It's 1 by default. If it's not zero, output tokens are randomly sampled from the logits, so the one with the highest log probability is not necessarily selected.
